pkg/server/plugin/keymanager/azurekeyvault/azure_key_vault.go

```diff
@@ -171,17 +171,20 @@ func (p *Plugin) Configure(ctx context.Context, req *configv1.ConfigureRequest)
 		if err != nil {
 			return nil, status.Errorf(codes.Internal, "failed to create Key Vault client with client credentials: %v", err)
 		}
+
 	case config.UseMSI:
-		cred, err := p.hooks.msiCredential()
+		p.log.Warn("use_msi is deprecated and will be removed in a future release")
+		fallthrough // use default credential which attempts to fetch credentials using MSI
+
+	default:
+		cred, err := azidentity.NewDefaultAzureCredential(nil)
```

Comment:
can you add a test case for this? 
and may you remove the hook? or replace it with a new one but to replace direct use of azidentityNewDefaultAzureCredential

Reply:
I have modified the existing test. Otherwise, it's just an API wrapper from before.